apps/api/src/app/shared/utils/template-parser/new-liquid-parser.ts

```diff
@@ -788,7 +788,7 @@ function extractVariablesFromCondition(condition: string): string[] {
   processedCondition = processedCondition.replace(/\|\s*[a-zA-Z_][a-zA-Z0-9_]*(?:\s*:[^|%}]*)?/g, '');
 
   // Now match variable patterns from the processed condition
-  const variableMatches = processedCondition.match(/[a-zA-Z_][a-zA-Z0-9_]*(?:\.[a-zA-Z_][a-zA-Z0-9_]*)*(?:\[\d+\])*/g);
+  const variableMatches = processedCondition.match(/[a-zA-Z_][a-zA-Z0-9_[\].-]+/g);
```

Comment:
More permissive regex, since we later have a validation of extracted variables against schema, its safer approach to rather extract even potentially invalid ones like "steps.events..[abc]" and then disallow them later as opposed to do complicated regex and risk missing extraction of even valid ones.